internal/api/http/controller/registration.go

```diff
@@ -0,0 +1,34 @@
+package controller
+
+import (
+	"net/http"
+
+	"go.opentelemetry.io/otel/trace"
+
+	"GoEdu/internal/api/http/server"
+	"GoEdu/internal/pkg/logger"
+)
+
+type registrationHTTPController struct {
+	tracer trace.Tracer
+}
+
+func newRegistrationHTTPController(t trace.Tracer) *registrationHTTPController {
+	return &registrationHTTPController{tracer: t}
+}
+
+func (c *registrationHTTPController) RegistrationNew(w http.ResponseWriter, r *http.Request) {
+	logger.FromContext(r.Context()).Info("registration new")
+	w.WriteHeader(http.StatusNotImplemented)
```

Comment:
is it an error like a 404?

Reply:
See [501 status code](https://datatracker.ietf.org/doc/html/rfc7231#section-6.6.2).

In other words, that status code signals that logic for this API-endpoint hasn't been implemented yet. We will populate the call to registration module as soon as contract will be ready.